assets/auto-vj-uri.html

```diff
@@ -0,0 +1,141 @@
+            function tryCreatePlayer(meta) {
+                var vLayer = document.createElement('video');
+                vLayer.muted = true;
+                vLayer.autoplay = false;
+                vLayer.defaultPlaybackRate = meta.tempo;
+                vLayer.playbackRate = meta.tempo;
+                vLayer.addEventListener('loadeddata', () => {
+                    if(vLayer.readyState == HTMLMediaElement.HAVE_ENOUGH_DATA && players[meta.filePath] && !vLayer.didStart) { // can play 100% and not removed yet
+                        document.getElementById('container').appendChild(vLayer);
+                        vLayer.play();
+                        vLayer.didStart = true;
+                    }
+                });
+                vLayer.addEventListener('ended',() => {
+                    vLayer.style.display = "none";
+                },false);
+                
+                var uReader = new XMLHttpRequest();
+                uReader.open('GET', getFilename(meta));
+                uReader.responseType = 'text'
+                uReader.onload = () => {
+                    videofilename = LOCAL_URI + uReader.responseText;
+                    const extensions = [".webm", ".mp4"];
+                    for (i = 0; i < extensions.length; i++) {
+                        videopath = videofilename + extensions[i];
+                        console.log("Searching for video file: " + videopath);
+                        if(doesFileExist(videopath)){
+                            vLayer.src = videopath;
+                            break;
+                        }       
+                    }
+                };                                                                                                                                                                                                                                                                                                                                                                                                                                           
+                uReader.send();
+                return vLayer;
+            }
+
+            // Check if URI file exist
+            function doesFileExist(urlToFile)
+            {
+                var xhr = new XMLHttpRequest();
+                xhr.open('HEAD', urlToFile, false);
+                xhr.send();
+                if (xhr.readyState == 4 && xhr.status == 404 ) {
+                    console.log("File doesn't exist: " + urlToFile);
+                    return false;
+                } else {
+                    console.log("File found: " + urlToFile);
+                    return true;
+                }
+            }
+
+            // Called when track ends playing
+            function popTrack(meta) {
+                if(players[meta.filePath]) {
+                    players[meta.filePath].remove();
+                    delete players[meta.filePath];
+                }
+            }
```

Comment:
After extensive testing I have a big issue with this.

Recreating the video element every time the "source" changes it sometimes lags really hard or freezes the browser for several seconds, especially when switching decks often or fast back and forth.

I'm writing an alternative HTML version with my following requirements:
- Video element is handled by "meta.deck" instead of "meta.filePath".
- A video element is created per deck when the first track is pushed.
- The video element is never deleted and the video source is changed instead.
- Track the ticking of each "songsOnAir" and not just the "tickedDeck".
- Keep the "tickedDeck" video element above others.
- Smoother transition.

Reply:
> The original one deletes the element in concern of saving RAM so there's that. If you need instant switchovers then this is not the right approach of course. Maybe an OBS plugin that plugs into this by Websocket could be a better idea for such use cases. (But not as part of this project to avoid GPL)

This is what I thought for the first time too! But it would take way more effort and time, which I currently don't have.
The RAM tradeoff will work for me as of now.
Anyway this can go into another PR, leaving the comment in here just for informational purposes.

Reply:
OK, looking forward to the new thing later!